vspace/src/main/java/edu/asu/diging/vspace/web/staff/UpdateContentOrderController.java

```diff
@@ -23,17 +30,25 @@ public class UpdateContentOrderController {
 
     @Autowired
     private IContentBlockManager contentBlockManager;
+    
+    @Autowired
+    private ISlideManager slideManager;
 
     @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/blocks/order/update", method = RequestMethod.POST)
-    public ResponseEntity<List<ContentBlock>> adjustContentOrder(@RequestBody List<ContentBlock> contentBlockList) {
+    public ResponseEntity<List<ContentBlock>> adjustContentOrder(@RequestBody String contentBlockString, @PathVariable("id") String slideId) {
 
+        ObjectMapper objectMapper = new ObjectMapper();
+        ISlide slide = slideManager.getSlide(slideId);
+        List<ContentBlock> contentBlockList;
         try {
-            contentBlockManager.updateContentOrder(contentBlockList);
-        } catch (BlockDoesNotExistException e) {
-            logger.warn("Block Id does not exist, bad request.", e);
-            return new ResponseEntity<List<ContentBlock>>(HttpStatus.BAD_REQUEST);
+            System.out.println(contentBlockString);
+            contentBlockList = objectMapper.readValue(contentBlockString, new TypeReference<List<ContentBlock>>(){});
+            contentBlockManager.updateContentOrder(contentBlockList,slide);
+        } catch (Exception e) {
```

Comment:
exceptions should be specific or is `Exception` really thrown here (in this case, there should be a comment here).